src/Main.elm

```diff
@@ -0,0 +1,375 @@
+                                    ]
+                            )
+                    )
+            )
+
+
+view : Model -> Html Msg
+view model =
+    div [ css [ mainStyle ] ]
+        [ if model.toggleModal then
+            modal model.boardSize ToggleModal DecCell AddCell
+
+          else
+            div [] []
+        , div [ css [ titleStyle ] ] [ text <| viewStatus model.gameState model.currentPlayer ]
+        , div [] <| viewBoard model.boardSize model.board
+        , case model.gameState of
+            Started ->
+                div [] []
+
+            Finished _ ->
+                button [ css [ playAgainButtonStyle ], onClick PlayAgain ] [ text "Play Again" ]
+        ]
+
+
+
+-- Reusable Function
+
+
+indices : Model -> List Int
+indices model =
+    List.range 0 (model.boardSize - 1)
+
+
+
+-- Function to get a row from the board
+
+
+row : Model -> b -> List ( Int, b )
+row model y =
+    indices model
+        |> List.map (\x -> ( x, y ))
+
+
+
+-- Function to get a column from the board
+
+
+collumn : Model -> b -> List ( b, Int )
+collumn model x =
+    indices model
+        |> List.map (\y -> ( x, y ))
+
+
+
+-- Function to get the diagonal from the board (top left to bottom right)
+
+
+diagonal1 : Model -> List ( Int, Int )
+diagonal1 model =
+    indices model
+        |> List.map (\i -> ( i, i ))
+
+
+
+-- Function to get the diagonal from the board (top right to bottom left)
+
+
+diagonal2 : Model -> List ( Int, Int )
+diagonal2 model =
+    indices model
+        |> List.map (\i -> ( i, model.boardSize - 1 - i ))
+
+
+
+-- Function to get all the winning positions
+
+
+winPositions : Model -> List (List ( Int, Int ))
+winPositions model =
+    (indices model
+        |> List.map (\y -> row model y)
+    )
+        ++ (indices model
+                |> List.map (\x -> collumn model x)
+           )
+        ++ [ diagonal1 model, diagonal2 model ]
+
+
+
+-- Function to check if all elements in a list are equal
+
+
+allEqual : Board -> List ( Int, Int ) -> Maybe Player
+allEqual board l =
+    case l of
+        [] ->
+            Nothing
+
+        h :: t ->
+            if List.all (\x -> Dict.get h board == Dict.get x board) t then
+                Dict.get h board
+
+            else
+                Nothing
+
+
+
+-- Function to check if a square in the board is empty (i.e. contains 0)
+
+
+isEmpty : Board -> ( Int, Int ) -> Bool
+isEmpty board ( x, y ) =
+    case Dict.get ( x, y ) board of
+        Just _ ->
+            False
+
+        Nothing ->
+            True
+
+
+
+-- Function to check if a player has won the game
+
+
+checkWinner : Board -> Model -> Maybe Player
+checkWinner board model =
+    winPositions model
+        |> List.map (allEqual board)
+        |> List.filter
+            (\x ->
+                case x of
+                    Just _ ->
+                        True
+
+                    _ ->
+                        False
+            )
```

Comment:
Just a nitpick: You could use `Maybe.isJust` (but that requires `maybe-extra` package). You could also just replace `List.map f >> List.filter Maybe.isJust` with `List.filterMap f`